staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -34,8 +34,10 @@ type CustomResourceDefinitionSpec struct {
 	// Scope indicates whether this resource is cluster or namespace scoped.  Default is namespaced
 	Scope ResourceScope
 	// Validation describes the validation methods for CustomResources
+	// Optional, and correspond to the first version in the versions list
 	Validation *CustomResourceValidation
 	// Subresources describes the subresources for CustomResources
+	// Optional, and correspond to the first version in the versions list
 	Subresources *CustomResourceSubresources
```

Comment:
don't we have some syntax for optional fields? `+optional+` or something like that

Reply:
`+optional` tag causes openapi spec to reflect that the field is optional. In openapi spec we only populate versioned types, so having `+optional` tag in internal types doesn't change openapi spec. But we can have `+optional` tag in internal types for documentation/symmetry purpose.

I wonder what if we have multiple external versions, and one field becomes optional in a newer version. Does the internal type always reflect the newer version and mark the field as optional?

Reply:
The internal type has to be able to express this at least. So documenting it there makes sense IMO.